### PR TITLE
Vander extension

### PR DESCRIPTION
**Pull request by nassimberrada**

### CODE REVIEW

**Review by Ishticode — CHANGES_REQUESTED** [review]
Looks mostly good. Two requests
Could you add the container (instance and static) and array methods
Could you move the example of container from ivy functional over to the container method.
The `NativeArray` examples are not required.
This would be the format with most functions, an array instance method, a container instance method and a container static method would be required usually. These methods will have their own docstring and examples. The examples of these methods are not to be added in ivy functional. Sorry if there is repetition. Please feel free to raise any question Nassim. This is good work. Thank you 🙂

**Comment by nassimberrada** [comment]
Not at all, i just wasn't sure this structure was the same for all functions, thank you for bearing with me !

File: ivy/array/linear_algebra.py [hunk 1]
```diff
@@ -453,3 +453,68 @@ def vector_to_skew_symmetric_matrix(
         self: ivy.Array, *, out: Optional[ivy.Array] = None
     ) -> ivy.Array:
         return ivy.vector_to_skew_symmetric_matrix(self._data, out=out)
+
+    def vandedr(
```
Comment (Ishticode) [inline ivy/array/linear_algebra.py:457]: name type

File: ivy/array/linear_algebra.py [hunk 2]
```diff
@@ -453,3 +453,68 @@ def vector_to_skew_symmetric_matrix(
...
+        """
+        ivy.Array instance method variant of ivy.vander.
+        This method Returns the Vandermonde matrix of the input array.
+
+        Parameters
+        ----------
+        self
+            1-D input array.
+        N
+         Number of columns in the output. If N is not specified,
```
Comment (Ishticode) [inline ivy/array/linear_algebra.py:474]: explanation requires indenting

File: ivy/container/linear_algebra.py [hunk 3]
```diff
@@ -2009,3 +2009,148 @@ def vector_to_skew_symmetric_matrix(
...
+                    [343,  49,   7,   1],
+                    [512,  64,   8,   1],
+                    [729,  81,   9,   1]]
+                    )
+        }
+        """
+        return ContainerBase.multi_map_in_static_method(
+            "vander",
+            x,
+            N=N,
```
Comment (Ishticode) [inline ivy/container/linear_algebra.py:2073]: could you copy all the kwargs here as we did in `flatten`

File: ivy/container/linear_algebra.py [hunk 4]
```diff
@@ -2009,3 +2009,148 @@ def vector_to_skew_symmetric_matrix(
...
+        ivy.Container instance method variant of ivy.vander.
+        This method Returns the sum along the specified diagonals of a matrix (or a
+        stack of matrices).
+
+        Parameters
+        ----------
+        self
+            input container having shape ``(..., M, N)`` and whose innermost two
+            dimensions form ``MxN`` matrices. Should have a floating-point data type.
+        offset
```
Comment (Ishticode) [inline ivy/container/linear_algebra.py:2096]: N and increasing parameters not explained here

File: ivy/container/linear_algebra.py [hunk 5]
```diff
@@ -2009,3 +2009,148 @@ def vector_to_skew_symmetric_matrix(
...
+        self: ivy.Container,
+        /,
+        *,
+        N: Optional[int] = None,
+        increasing: Optional[bool] = False,
+        out: Optional[ivy.Container] = None,
+    ) -> ivy.Container:
+        """
+        ivy.Container instance method variant of ivy.vander.
+        This method Returns the sum along the specified diagonals of a matrix (or a
```
Comment (Ishticode) [inline ivy/container/linear_algebra.py:2088]: this seems like an explanation of the trace function. Maybe you forgot to update the docstring here 😂

File: ivy/container/linear_algebra.py [hunk 6]
```diff
@@ -2009,3 +2009,148 @@ def vector_to_skew_symmetric_matrix(
...
+
+        Examples
+        --------
+        With :class:`ivy.Container` inputs:
+
+        >>> x = ivy.Container(
+                a = ivy.array([1, 2, 3, 5])
+                b = ivy.array([6, 7, 8, 9])
+            )
+        >>> ivy.vander(x)
```
Comment (Ishticode) [inline ivy/container/linear_algebra.py:2135]: instance mehtod call should be x.vander()

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi. Looking good. Some minor typos and mistakes. I have made inline comments. Pls feel free to query further or correct me if you believe I hv made a mistake anywhere. When done, pls ensure the tests are passing and hopefully this can be merged in.

**Comment by nassimberrada** [comment]
I did forget to modify the docstrings after copying them didn't i hahah, i corrected that as fast as i could type

**Comment by nassimberrada** [comment]
I did forget to modify the docstrings after copying them didn't i hahah, i corrected that as fast as i could type

**Comment by nassimberrada** [comment]
I made sure the tests pass on all four backends now, let me know if there's anything else that requires changing

File: ivy/container/linear_algebra.py [hunk 7]
```diff
@@ -2009,3 +2009,138 @@ def vector_to_skew_symmetric_matrix(
...
+                    )
+        }
+        """
+        return ContainerBase.multi_map_in_static_method(
+            "vander",
+            x,
+            key_chains,
+            to_apply,
+            prune_unapplied,
+            map_sequences,
```
Comment (Ishticode) [inline ivy/container/linear_algebra.py:2080]: perhaps pass this as keyword arguments

**Review by Ishticode — CHANGES_REQUESTED** [review]
ig should be ready to merge after these

**Review by Ishticode — APPROVED** [review]
Looking great. Thanks Nassim.

**Comment by nassimberrada** [comment]
closes #5562

**Pull request merged**